googlemock/CMakeLists.txt.install

```diff
@@ -16,6 +16,16 @@ add_library(gmock STATIC
   ${gtest_vendor_BASE_DIR}/src/gtest-all.cc
   src/gmock-all.cc)
 target_link_libraries(gmock ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
Is `CMAKE_THREAD_LIBS_INIT` still needed? Or does the explicit linking against `pthread` cover all the cases?

Reply:
`-pthread` covers it on Linux, but I don't know about Windows and macOS.  I think the worst case here on Linux is that `-pthread` gets added twice, which isn't a huge deal, so I think it would be better to be safe here and leave `CMAKE_THREAD_LIBS_INIT`.

Reply:
> I don't know about Windows and macOS. I think the worst case here on Linux is that -pthread gets added twice, which isn't a huge deal

I would rather check how `CMAKE_THREAD_LIBS_INIT` behaves across platforms and then make a decision based on that knowledge:

* macOS: https://ci.ros2.org/job/ci_osx/4927/consoleFull#console-section-82 -> empty
* Windows: https://ci.ros2.org/job/ci_windows/5850/consoleFull#console-section-79 -> empty

As a consequence I think the `Threads` logic should be removed.

Reply:
Done in 8885a1b